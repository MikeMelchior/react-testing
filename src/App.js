import React, { useState } from "react";

const App = () => {
  const [header, setHeader] = useState('Tropical Tigers')

  const clickHandler = () => {
    setHeader('Wonderful Wombats')
  }

  return (
    <>
      <button
        type="button"
        onClick={clickHandler}
      >Click me!
      </button>
      <h1>{header}</h1>
    </>
  )
}

export default App;
