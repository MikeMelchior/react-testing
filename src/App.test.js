import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';



describe("App Component", () => {
  it('renders tropical tigers', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot()
  });

  it('renders wonderful wombats after button click', async () => {
    const user = userEvent.setup()

    render(<App />);
    const button = screen.getByRole('button', {name: 'Click me!'});

    await user.click(button)
    

    expect(screen.getByRole('heading').textContent).toMatch(/wonderful wombats/i)
  })
})
