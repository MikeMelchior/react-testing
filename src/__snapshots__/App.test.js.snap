// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders tropical tigers 1`] = `
<div>
  <button
    type="button"
  >
    Click me!
  </button>
  <h1>
    Tropical Tigers
  </h1>
</div>
`;
